rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -299,15 +300,12 @@ public static FlowNode getFlowEndNode(FlowExecution execution) {
     public static List<FlowNode> getChildNodes(final FlowNode parentNode) {
         final List<FlowNode> nodes = new ArrayList<FlowNode>();
 
-        FlowNodeNavigator nodeNavigator = new FlowNodeNavigator(new FlowNodeNavigationListener() {
-            @Override
-            public void onNode(FlowNode navNode) {
-                if (navNode.getParents().contains(parentNode) && !nodes.contains(navNode)) {
-                    nodes.add(navNode);
-                }
+        FlowGraphWalker walker = new FlowGraphWalker(parentNode.getExecution());
+        for (FlowNode node : walker) {
+            if (node.getParents().contains(parentNode) && !nodes.contains(node)) {
+                nodes.add(node);
```

Comment:
@jglick Sorry ... I don't see how it's simplified ... looks like the same code. Only diff I can see is `List` Vs `Set`.

The `HashSet` might be a bit more optimised, but I wasn't going to assume anything wrt hashing/equals impls + I would think that the lists are so small that the performance diff would be near zero. Maybe those assumptions are not right though.